proto/src/determined/api/v1/model.proto

```diff
@@ -76,6 +76,15 @@ message GetModelsResponse {
   Pagination pagination = 2;
 }
 
+// Get a list of model labels.
+message GetModelLabelsRequest {}
+
+// Response to GetModelLabelsRequest.
+message GetModelLabelsResponse {
+  // List of labels used across all models.
+  repeated string labels = 1;
```

Comment:
The code looks good, but I am mildly concerned about extensibility of this. 
e.g. right now it's a list of plain strings, but I can see a case for returning a list of `[{'label': <label name>, 'count': <number of objects with the label>}]`, e.g. for faceted search UX. 
It'd be a breaking change if you go from list of strings -> list of objects, but won't be a problem if you start with objects.